hls4ml/utils/config.py

```diff
@@ -50,10 +57,265 @@ def _get_precision_from_quantizer(quantizer):
+    Instead, for each applicable layer, it uses data types set in the HLSModel config along with information about the
+    layer shape.
+
+    The function supports Dense-type layers only. If your model contains other layers with accum_t (e.g. convolutional
+    layers), consider using set_data_types_from_keras_model() instead.
+
+    Args:
+        config (dict): HLSModel configuration dictionary to be updated. Its granularity must be 'name'.
+        model: Keras model to be used for adjusting accum_t in relevant layers.
+
+    Returns:
+        None. The function makes changes directly to the supplied config.
+    """
+    if 'LayerName' not in config:
+        raise RuntimeError("The granularity of the supplied config is not 'name'.")
+
+    def find_optimal_a_b(max_val, min_val):
+        max_bits = 64
+
+        for a in range(1, max_bits + 1):
+            for b in range(0, a + 1):
+                max_possible = 2 ** b - 2 ** (b - a)
+                min_possible = 2 ** (b - a)
+
+                if min_possible <= min_val and max_possible >= max_val:
+                    return a + 1, b + 1
+
+        return None, None
+
+    for i, layer in enumerate(model.layers):
+        name = layer.name
+
+        if name not in config['LayerName']:
+            print(f"accum_t profiling: {name} not present in config['LayerName'], ignoring.")
+            continue
+
+        if 'accum' not in config['LayerName'][name]['Precision']:
+            continue
+
+        if not isinstance(layer, keras.layers.Dense):
+            print(f"accum_t profiling: {name} is not a Dense layer, ignoring as only Dense layers are currently "
+                  "supported. You can use set_data_types_from_keras_model() instead.")
+            continue
+
+        type_w = VivadoBackend.convert_precision_string(None, config['LayerName'][name]['Precision']['weight'])
+        type_b = VivadoBackend.convert_precision_string(None, config['LayerName'][name]['Precision']['bias'])
+
+        if i == 0:
+            previous_layer_config = config['LayerName'][name + '_input']
+        else:
+            previous_layer_config = config['LayerName'][model.layers[i - 1].name]
+
+        if isinstance(previous_layer_config['Precision'], dict):
+            type_i = VivadoBackend.convert_precision_string(None, previous_layer_config['Precision']['result'])
+        else:
+            type_i = VivadoBackend.convert_precision_string(None, previous_layer_config['Precision'])
+
+        # Assuming that all of type_w, type_b and type_i are FixedPrecisionType objects.
+        n = layer.output_shape[1]
+        max_w = 2 ** (type_w.integer - 1) - 2 ** (type_w.integer - type_w.width)
+        max_i = 2 ** (type_i.integer - 1) - 2 ** (type_i.integer - type_i.width)
+        max_b = 2 ** (type_b.integer - 1) - 2 ** (type_b.integer - type_b.width)
+        min_b = 2 ** (type_b.integer - type_b.width)
+
+        max_val = n * max_w * max_i + max_b
+        min_val = min_b
```

Comment:
I think in the general case the minimum value could come either from the product of weight * input or the bias, depending on their precisions, not always from the bias.

Reply:
bias may not exist (e.g., `Dense(..., use_bias=False)`), in that case we insert zeros and set precision to uint(1), which may throw this function off.

Reply:
I've just pushed the commits addressing both issues.